### DISCUSSION TITLE
Is address required when referring to other contracts?

### DISCUSSION THREAD

**Discussion by nigeltims** [smartcontractkit/full-blockchain-solidity-course-py#402] [Q&A] [answered]
Hi, I was doing lesson 2: Storage Factory and I tried to implement sfGet() on my own. I ended up writing 

        return simpleStorageArray[_simpleStorageIndex].retrieve();

Is this correct? I don't use the address() to access the specific SimpleStorage contract at the index in the array, but when I test it seems to work correctly.

**Comment by cromewar** [comment, marked as answer]
Hello @nigeltims well there isn't just one way to code, or to resolve problems so if this worked for you, that's completely ok.
However is also true that is best practice to access directly to the addresses rather than indexes on arrays, if you are interested about good practices with solidity, please read [this](https://medium.com/newcryptoblock/best-practices-in-solidity-b324b65d33b1).

Best Regards, Cromewar.

**Reply by nigeltims** [reply]
Hi @cromewar Thank you so much for your response, I will go forward using the addresses.
